aws_xray_sdk/core/patcher.py

```diff
@@ -80,3 +110,83 @@ def _patch(module_to_patch):
 
     _PATCHED_MODULES.add(module_to_patch)
     log.info('successfully patched module %s', module_to_patch)
+
+
+def _patch_func(parent, func_name, func, modifier=lambda x: x):
+    from aws_xray_sdk.core import xray_recorder
+
+    capture_name = func_name
+    if func_name.startswith('__') and func_name.endswith('__'):
+        capture_name = '{}.{}'.format(parent.__name__, capture_name)
+    setattr(parent, func_name, modifier(xray_recorder.capture(name=capture_name)(func)))
+
+
+def _patch_class(module, cls):
+    for member_name, member in inspect.getmembers(cls, inspect.isclass):
+        if member.__module__ == module.__name__:
+            # Only patch classes of the module, ignore imports
+            _patch_class(module, member)
+
+    for member_name, member in inspect.getmembers(cls, inspect.ismethod):
+        if member.__module__ == module.__name__:
+            # Only patch methods of the class defined in the module, ignore inherited
+            _patch_func(cls, member_name, member)
+
+    for member_name, member in inspect.getmembers(cls, inspect.isfunction):
+        if member.__module__ == module.__name__:
+            # Only patch static methods of the class defined in the module, ignore inherited
+            if is_instance_method(cls, member_name, member):
+                _patch_func(cls, member_name, member)
+            else:
+                _patch_func(cls, member_name, member, modifier=staticmethod)
+
+
+def _on_import(module):
+    for member_name, member in inspect.getmembers(module, inspect.isfunction):
+        if member.__module__ == module.__name__:
+            # Only patch functions of the module, ignore imports
+            _patch_func(module, member_name, member)
+
+    for member_name, member in inspect.getmembers(module, inspect.isclass):
+        if member.__module__ == module.__name__:
+            # Only patch classes of the module, ignore imports
+            _patch_class(module, member)
+
+
+def _external_module_patch(module, ignore_module_patterns):
+    if module.startswith('.'):
+        raise Exception('relative packages not supported for patching: {}'.format(module))
+
+    if module in _PATCHED_MODULES:
+        log.debug('%s already patched', module)
+    elif any(pattern.match(module) for pattern in ignore_module_patterns):
```

Comment:
Minor improvement could be stopping after you find the first matching pattern

Reply:
I believe that's what `any` internally does https://docs.python.org/3/library/functions.html#any

Reply:
but all patterns have to be checked first